### PR TITLE
feat: Added support for text/yaml in OpenAPI to be used in benchling-api-client BNCH-37249

### PR DESCRIPTION
**Pull request by miggyst**
Added text/yaml support.
Fixed some unit tests and added 1 to check for text/yaml support

### CODE REVIEW

**Review by bowenwr — CHANGES_REQUESTED** [review]
See https://github.com/benchling/benchling-api-client/pull/21#pullrequestreview-941745996

File: openapi_python_client/parser/responses.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ class Response:
     "application/json": "response.json()",
     "application/octet-stream": "response.content",
     "text/html": "response.text",
+    "text/yaml": "yaml.safe_load(response.text.encode('utf-8'))",
```
Comment (bowenwr) [inline openapi_python_client/parser/responses.py:25]: That's a lot of logic to put in the value - is it possible to encapsulate most of that somewhere into a function that accepts `response`?

```suggestion
    "text/yaml": "to_yaml(response)",
```

**Review by bowenwr — COMMENTED** [review]
I think this looks pretty good but added @daniel-deutsch-benchling to help make sure I'm not missing anything.

I did find some issues in https://github.com/benchling/benchling-api-client/pull/21 with repeated types that may need to be addressed here.

File: openapi_python_client/parser/responses.py (hunk 1 above)
Comment (miggyst) [reply openapi_python_client/parser/responses.py]: Broke this down on the latest commit

File: openapi_python_client/templates/property_templates/model_property.pyi [hunk 2]
```diff
@@ -1,6 +1,12 @@
 {% macro construct(property, source, initial_value=None) %}
 {% if property.required and not property.nullable %}
+{% if source == "response.yaml" %}
+yaml = YAML(typ="base")
```
Comment (bowenwr) [inline openapi_python_client/templates/property_templates/model_property.pyi:4]: We were previously using the `safe` loaded from PyYaml. Does that not work here?

```suggestion
yaml = YAML(typ="safe")
```

**Review by bowenwr — APPROVED** [review]
Prefer safe loader over base if it works. Otherwise, that's the last thing and I'm not going to block.

**Pull request merged**